First week's task

balazs482 opened the pull request:
The first week's task is executed on the init branch!

- Two classes and the main() function is added.
- The input from the example produces the expected output via manual testing.

balazs482 committed:
Creating Character class

Character class with HP, DMG and name variables

balazs482 committed:
Implementing Character class functions

Basic constructor, getter and setter functions

balazs482 committed:
Creating Words class

The Words class is used to create a vector of strings from a string by splitting at whitespaces

balazs482 committed:
Implementing Words class functions

Splitting happens inside the constructor, and the basic getter is done.

balazs482 committed:
Creating main() function

Running manual character creation and automatic fighting simulation. Reading/writing from/to the console.

balazs482 committed:
Now contains program description

Provides instructions for the usage and information on the operation of the program.

gboldi19 committed:
Basic formatting related to markdown synthax

gboldi19 committed:
Minor fixes to grammar, understandability

balazs482 committed:
Variable namechange from healthPoints to hitPoints

My life has been a lie all along...

balazs482 committed:
Variable namechange from healthPoints to hitPoints

Wikipedia says it's "hit points"

balazs482 committed:
Merge pull request #1 from gboldi19/patch-1

Merging temporary branch to .init

hegyhati commented:
Msot csak ralestem, de osszekevered a parancssori argumentumot az STDIN-nel. Ezt tessek fixalni.

gboldi19 committed:
Input type changed from string vector to parametered arguments

Parmama committed:
Update README.md

Parmama committed:
More instructions added to the description

Parmama committed:
Minor changes

Parmama committed:
Removed global scope std

Parmama committed:
Removed global scope std

gboldi19 committed:
Applied the previous changes to Character.cpp

gboldi19 committed:
Fixed a type-o

gboldi19 committed:
Description beautified

gboldi19 committed:
Description beautified

vizvezetek commented on Character.h:
```diff
@@ -0,0 +1,17 @@
+
+class Character
+{
+private:
+	std::string name;
+	unsigned int HP, DMG;
+public:
+	Character(std::string characterName, int hitPoints, int damagePoints);
+	void setHP(int hitPoints);
+	void setDMG(int damagePoints);
+	int getHP();
+	int getDMG();
+	void setName(std::string characterName);
+	std::string getName();
```
Ahol lehet, ott használjunk const adattagokat és metódusokat is. Setter-ek használata nélkül kell megoldani ezt a feladatot.

balazs482 replied:
@vizvezetek A settereket hogyan mellőzzük?  

1. Public-ba kerüljenek a változtatandó értékek
2. Készítsünk az osztályon belül függvényt a támadásra
3. Hozzon létre a main minden körben újabb példányt az előzőből másolt, valamint az újonnan számolt értékekkel
4. Egyéb

vizvezetek replied:
1. Ha Publicba kerül egy adattag, akkor azzal felborítod az OO alapelveket. Természetesen van helyzet, amikor ezt a módszert kell allkalmazni, de jelenleg nem.
2. Kiváló ötlet!
3. Az kicsit túbonyolított, és nem túl elegáns megoldás.
4. -

vizvezetek commented on Words.cpp:
```diff
@@ -0,0 +1,27 @@
+#include "Words.h"
+
+Words::Words(string input)
+{
+    string word = "";
+    for (int i = 0; i < input.length(); i++)
+    {
+        if (input[i] != ' ')
+        {
+            word += input[i];
+        }
+        else if (word != "")
+        {
+            wordVector.push_back(word);
+            word = "";
+        }
+    }
+    if (word != "")
+    {
+        wordVector.push_back(word);
+    }
+}
+
+string Words::getWord(int pos)
+{
+    return wordVector[pos];
+}
```
Ennek az osztálynak nem érzem a szükségességét. Mondhatjuk úgy is hogy felesleges.

vizvezetek reviewed (commented):
Kis hibák javítását követően én jónak gondolom. A bin file-t pedig nem kell feltölteni, erre van a .gitignore.

balazs482 committed:
Delete Words.h

Words class is no longer used.

balazs482 committed:
Delete Words.cpp

Words class is no longer used.

gboldi19 committed:
.gitignore added with basic rules

gboldi19 committed:
Fixes according to .gitignore rules

gboldi19 committed:
Setter, getter removal, DMG converted to static

gboldi19 committed:
Fixed minor issues

gboldi19 committed:
Beautified

balazs482 committed:
Variable type changes

Changed intigers to unsigned. Changed the cycle tracking intiger in main to bool. All variables, arguments and functions changed to constants (where possible).

gboldi19 committed:
Minor type-o as usual

balazs482 committed:
Unnecessary file deleted

Uploaded because of a temporarily incorrect .gitignore.

balazs482 commented:
@hegyhati megcsinaltuk a kert modositasokat

vizvezetek commented on Character.h:
```diff
@@ -0,0 +1,15 @@
+
+class Character
+{
+private:
+	std::string name;
+	unsigned int HP;
+    unsigned int DMG;
```
Ahogy korábban mondtam, ami lehet const, az legyen is az.

vizvezetek reviewed (commented):
Kis hiba kivételével rendben.

Parmama commented:
@vizvezetek megcsinaltuk a kert modositasokat

balazs482 committed:
Some character class are now constants

name and DMG values are now constants and the constructor uses initializing list to set them.

vizvezetek reviewed (approved):
@hegyhati

balazs482 commented:
@vizvezetek 
Kérlek review-olnád a PR-t? A már approve-olt változtatások óta nem történt módosítás.
Nem akarok sürgető üzeneteket írni, csak el szeretném kerülni, hogy elvesszen a sok között. Előre köszönjük!

vizvezetek reviewed (approved):
Én rendben gondolom. @hegyhati
